python/feathub/tests/feathub_it_test_base.py

```diff
@@ -106,6 +106,38 @@ def create_file_source(
             timestamp_format=timestamp_format,
         )
 
+    # TODO: only invoke the corresponding base class's setUpClass()
+    #  method to reduce resource consumption.
+    @classmethod
+    def invoke_all_base_class_setupclass(cls):
+        for base_class in cls.__bases__:
+            if issubclass(base_class, unittest.TestCase):
+                base_class.setUpClass()
+
+    @classmethod
+    def invoke_all_base_class_teardownclass(cls):
+        for base_class in cls.__bases__:
+            if issubclass(base_class, unittest.TestCase):
+                base_class.tearDownClass()
+
+    def invoke_base_class_setup(self):
+        self._get_base_test_class().setUp(self)
+
+    def invoke_base_class_teardown(self):
+        self._get_base_test_class().tearDown(self)
+
+    def _get_base_test_class(self) -> Type[unittest.TestCase]:
+        for base_class in self.__class__.__bases__:
+            method_list = [
+                func for func in dir(base_class) if callable(getattr(base_class, func))
```

Comment:
Instead of enumerating all methods in all base classes, is it possible to use `unittest.TestCase.id()` to get the base class name?

If it does not work, can we setup a hash table to map test name to base class name, instead of enumerating all methods for each test?

Reply:
`unittest.TestCase.id()` would return the current test class instead of the base class containing the current test method, thus we cannot directly use it in this situation. I'll add a hash table to hold the mapping of test names to base class names.